tuf/formats.py

```diff
@@ -474,227 +467,102 @@ def make_signable(role_schema):
-    self.info = {}
-    self.info['version'] = version
-    self.info['expires'] = expires
-    self.info['targets'] = filedict
-    self.info['delegations'] = delegations
 
 
-  @staticmethod
-  def from_metadata(targets_metadata):
-    # Is 'targets_metadata' a Targets metadata file?
-    # Raise securesystemslib.exceptions.FormatError if not.
-    tuf.formats.TARGETS_SCHEMA.check_match(targets_metadata)
 
-    version = targets_metadata['version']
-    expires = targets_metadata['expires']
-    filedict = targets_metadata.get('targets')
-    delegations = targets_metadata.get('delegations')
 
-    return TargetsFile(version, expires, filedict, delegations)
 
 
-  @staticmethod
-  def make_metadata(version, expiration_date, filedict=None, delegations=None):
-    if filedict is None and delegations is None:
-      raise securesystemslib.exceptions.Error('We don\'t allow completely'
-        ' empty targets metadata.')
+  # Automatically provide certain schema properties if they are not already
+  # provided and are required in objects of class <schema>.
+  # This includes:
+  #   _type:        <securesystemslib.schema.String object>
+  #   spec_version: SPECIFICATION_VERSION_SCHEMA
+  #
+  # (Please note that _required is slightly misleading, as it includes both
+  #  required and optional elements. It should probably be called _components.)
+  #
+  for schema_element in schema._required: #pylint: disable=protected-access
+    key = schema_element[0]
+    element_type = schema_element[1]
 
-    result = {'_type' : 'targets'}
-    result['spec_version'] = tuf.SPECIFICATION_VERSION
-    result['version'] = version
-    result['expires'] = expiration_date
-    result['targets'] = {}
+    if key in dictionary:
+      # If the field has been provided, proceed normally.
+      continue
 
-    if filedict is not None:
-      result['targets'] = filedict
-    if delegations is not None:
-      result['delegations'] = delegations
+    elif isinstance(element_type, SCHEMA.Optional):
+      # If the field has NOT been provided but IS optional, proceed without it.
+      continue
 
-    # Is 'result' a Targets metadata file?
-    # Raise 'securesystemslib.exceptions.FormatError' if not.
-    tuf.formats.TARGETS_SCHEMA.check_match(result)
+    else:
+      # If the field has not been provided and is required, check to see if
+      # the field is one of the one of the fields we automatically fill.
 
-    return result
+      # Currently, the list is limited to ['_type', 'spec_version'].
 
+      if key == '_type' and isinstance(element_type, SCHEMA.String):
+        # A SCHEMA.String stores its expected value in _string, so use that.
+        dictionary[key] = element_type._string #pylint: disable=protected-access
 
+      elif (key == 'spec_version' and
+          element_type == SPECIFICATION_VERSION_SCHEMA):
+        # If not provided, use the specification version in tuf/__init__.py
+        dictionary[key] = tuf.SPECIFICATION_VERSION
```

Comment:
I know, I [suggested this](https://github.com/awwad/tuf/pull/23#discussion_r269027063), but now that I look at it I think it's pretty specific, at least for the function name `build_dict_conforming_to_schema`. Do you think there is a better name? If not, disregard this comment.

Reply:
The function is still intended to be a general purpose (Object-type) schema builder, so I would rather remove this convenience functionality than change that scope.  There may be a better name, but I don't want to narrow its purpose to Role metadata construction, if that's what you mean.

Reply:
I think it's fine as it is.